core/trino-main/src/main/java/io/trino/execution/scheduler/StageTaskSourceFactory.java

```diff
@@ -543,8 +555,16 @@ public SourceDistributionTaskSource(
             this.splitBatchSize = splitBatchSize;
             this.getSplitTimeRecorder = requireNonNull(getSplitTimeRecorder, "getSplitTimeRecorder is null");
             this.catalogRequirement = requireNonNull(catalogRequirement, "catalogRequirement is null");
-            checkArgument(targetPartitionSplitCount > 0, "targetPartitionSplitCount must be positive: %s", targetPartitionSplitCount);
-            this.targetPartitionSplitCount = targetPartitionSplitCount;
+            checkArgument(targetPartitionSplitWeight > 0, "targetPartitionSplitCount must be positive: %s", targetPartitionSplitWeight);
+            this.targetPartitionSplitWeight = targetPartitionSplitWeight;
+            checkArgument(minPartitionSplitCount >= 0, "minPartitionSplitCount must be positive: %s", minPartitionSplitCount);
+            this.minPartitionSplitCount = minPartitionSplitCount;
+            checkArgument(maxPartitionSplitCount > 0, "maxPartitionSplitCount must be positive: %s", maxPartitionSplitCount);
```

Comment:
`%s` -> `%d`?

Reply:
Naa - it is common to use %s always in the codebase; even for integers (unless we need special formatting, like hex).